lib/api/core/models/repositories/repository.js

```diff
@@ -6,28 +6,39 @@ var
 
 /**
  * @param {Kuzzle} kuzzle
- * @param options
  * @constructor
  */
-function Repository (kuzzle, options) {
-  this.collection = options.collection;
-  this.index = options.index;
-  this.ttl = options.ttl || kuzzle.config.repositories.cacheTTL;
-  this.ObjectConstructor = options.ObjectConstructor;
-  this.writeLayer = options.writeLayer;
+function Repository (kuzzle) {
+  if (arguments.length === 0) { return; }
```

Comment:
OK, found it, I have a bad case on prototype inheritance.
with `RoleRepository.prototype = new Repository();`, nodejs calls the constructor without arguments when we declare the prototype.
Fixed with `RoleRepository.prototype = Object.create(Repository.prototype);` instead.